web/src/main/java/se/inera/intyg/webcert/web/csintegration/unit/CertificateServiceIntegrationUnitHelper.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2024 Inera AB (http://www.inera.se)
+ *
+ * This file is part of sklintyg (https://github.com/sklintyg).
+ *
+ * sklintyg is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sklintyg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.webcert.web.csintegration.unit;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import se.inera.intyg.infra.integration.hsatk.model.legacy.AbstractVardenhet;
+import se.inera.intyg.infra.security.common.model.IntygUser;
+
+@Component
+@RequiredArgsConstructor
+public class CertificateServiceIntegrationUnitHelper {
+
+    private final CertificateServiceVardenhetConverter certificateServiceVardenhetConverter;
+
+    public CertificateServiceUnitDTO getCareProvider(IntygUser user) {
```

Comment:
This file is a copy of the UnitHelper that uses WebcertUser which implement IntygUser. I think it would be better to reuse the code so we don't risk that the to helpers diverge

Reply:
Same for UserHelper

Reply:
In then first iteration I reused the code with this function:
```java
    private IntygUser getUser(Optional<IntygUser> optionalUser) {
        return optionalUser.orElseGet(webCertUserService::getUser);
    }
```

But from my understanding the request will diff more after we introduce more parameters related to the deep integrated user, therefor it was my understanding that it might be a good idea to separate them.

I will look into this and update if we can reuse more code!